common/src/main/java/com/discordsrv/common/feature/linking/LinkingRewardsModule.java

```diff
@@ -118,26 +135,41 @@ private boolean doesProfileAlreadyHave(ProfileImpl profile, RewardsConfig.Reward
         }
     }
 
-    private void addRewardToProfile(ProfileImpl profile, boolean game, RewardsConfig.Reward reward) {
+    /**
+     * Adds a reward to the profile's granted or pending rewards, depending on whether the reward is pending. And removes it from the opposite set.
+     *
+     * @param profile The profile to add the reward to.
+     * @param game    Whether the reward is for the game or Discord. True for game, false for Discord.
+     * @param reward  The reward to add.
+     * @param pending Whether the reward is pending or already granted.
+     */
+    private void addRewardToProfile(ProfileImpl profile, boolean game, RewardsConfig.Reward reward, boolean pending) {
+        Set<String> targetRewards;
+        Set<String> oppositeRewards;
+
         if (game) {
-            Set<String> gameRewards = profile.getGameGrantedRewards();
-            if (gameRewards == null) {
-                throw new IllegalStateException("Game profile not available");
-            }
-            gameRewards.add(reward.rewardId);
+            targetRewards = pending ? profile.getGamePendingRewards() : profile.getGameGrantedRewards();
+            oppositeRewards = pending ? profile.getGameGrantedRewards() : profile.getGamePendingRewards();
         } else {
-            Set<String> discordRewards = profile.getDiscordGrantedRewards();
-            if (discordRewards == null) {
-                throw new IllegalStateException("Discord profile not available");
-            }
-            discordRewards.add(reward.rewardId);
+            targetRewards = pending ? profile.getDiscordPendingRewards() : profile.getDiscordGrantedRewards();
+            oppositeRewards = pending ? profile.getDiscordGrantedRewards() : profile.getDiscordPendingRewards();
         }
+
+        if (targetRewards == null) {
+            throw new IllegalStateException((game ? "Game" : "Discord") + " profile not available");
+        }
+
+        targetRewards.add(reward.rewardId);
+        if (oppositeRewards == null) {
+            oppositeRewards = new HashSet<>();
+        }
+        oppositeRewards.remove(reward.rewardId);
     }
 
-    private void triggerRewards(ProfileImpl profile, RewardsConfig.LinkingReward.Type type) {
+    private void triggerRewards(ProfileImpl profile, com.discordsrv.common.config.main.RewardsConfig.LinkingReward.Type type) {
```

Comment:
I don't see a reason to use fully qualified class names here and below

Reply:
These are still fully qualified, ie. `com.discordsrv.common.config.main.RewardsConfig.LinkingReward.Type` instead of just `RewardsConfig.LinkingReward.Type`

Is there a reason they need to be fully qualified?